tf_agents/cogment_verse_tf_agents/reinforce/reinforce.py

```diff
@@ -133,12 +133,17 @@ def reset_replay_buffer(self):
         self._replay_buffer._write_index = -1
         self._replay_buffer._n = 0
 
-    def init_params(self, model_parms):
-        self._model.set_weights(model_parms)
-        self._model.trainable = True
-
     def get_replay_buffer_size(self):
         """
         Return the size of the internal replay buffer
         """
         return self._replay_buffer.size()
+
+    def save(self, f):
+        self._params["model_params"] = self._model.get_weights()
+        torch.save(self._params, f)
+        return {}
+
+    def load(self, f):
```

Comment:
This should be a static function that returns a new instance.

Reply:
because it's only ever used in this context by first creating an instance of the class and then calling load having everything in a static function would let load be a "factory" you wouldn't have to worry about "undoing" what's done in the constructor.

e.g. you could do something like that

```python
@staticmethod
def load(f):
    agent_params = pkl.load(f)
    return ReinforceAgent(**agent_params)
```